src/Sleep.js

```diff
@@ -4,78 +4,104 @@ class Sleep {
-    }, 0)
-    return totalSleepQuality / sleepData.length
+    var totalSleepQuality = this.getAverageOfValues(this.sleepData, "sleepQuality");
+    return totalSleepQuality;
   }
-  determineBestSleepers(date, userRepo) {
+
+  getUserSleepObject(date, userRepo, sleepProperty) {
     let timeline = userRepo.chooseWeekDataForAllUsers(this.sleepData, date);
-    let userSleepObject = userRepo.isolateUsernameAndRelevantData(this.sleepData, date, 'sleepQuality', timeline);
-
-    return Object.keys(userSleepObject).filter(function(key) {
-      return (userSleepObject[key].reduce(function(sumSoFar, sleepQualityValue) {
-        sumSoFar += sleepQualityValue
-        return sumSoFar;
-      }, 0) / userSleepObject[key].length) > 3
-    }).map(function(sleeper) {
+    let userSleepObject = userRepo.combineRankedUserIDsAndAveragedData(this.sleepData, date, sleepProperty, timeline);
+    return userSleepObject;
+  }
+
+  getUserNames(userSleepObject, userRepo) {
+    return Object.keys(userSleepObject).filter((key) => {
+      return this.getAverageOfValues(userSleepObject[key]) > 3;
+    }).map((sleeper) => {
       return userRepo.getDataFromID(parseInt(sleeper)).name;
     })
   }
-  determineSleepWinnerForWeek(date, userRepo) {
+
+  determineBestSleepers(date, userRepo) {
     let timeline = userRepo.chooseWeekDataForAllUsers(this.sleepData, date);
-    let sleepRankWithData = userRepo.combineRankedUserIDsAndAveragedData(this.sleepData, date, 'sleepQuality', timeline);
+    let userSleepObject = userRepo.isolateUsernameAndRelevantData(this.sleepData, date, "sleepQuality", timeline);
+    return this.getUserNames(userSleepObject, userRepo)
+  }
 
+  determineSleepWinnerForWeek(date, userRepo) {
+    let sleepRankWithData = this.getUserSleepObject(date, userRepo, "sleepQuality");
     return this.getWinnerNamesFromList(sleepRankWithData, userRepo);
   }
-  determineSleepHoursWinnerForDay(date, userRepo) {
-    let timeline = userRepo.chooseDayDataForAllUsers(this.sleepData, date);
-    let sleepRankWithData = userRepo.combineRankedUserIDsAndAveragedData(this.sleepData, date, 'hoursSlept', timeline);
 
+  determineSleepHoursWinnerForDay(date, userRepo) {
+    let sleepRankWithData = this.getUserSleepObject(date, userRepo, "hoursSlept")
+    console.log(sleepRankWithData);
     return this.getWinnerNamesFromList(sleepRankWithData, userRepo);
   }
-  getWinnerNamesFromList(sortedArray, userRepo) {
-    let bestSleepers = sortedArray.filter(function(element) {
+
+  getBestResults(sortedArray) {
+    return sortedArray.filter(element => {
       return element[Object.keys(element)] === Object.values(sortedArray[0])[0]
     });
+  }
 
-    let bestSleeperIds = bestSleepers.map(function(bestSleeper) {
+  getSleepersIds(bestSleepers) {
+    return bestSleepers.map(bestSleeper => {
       return (Object.keys(bestSleeper));
     });
+  }
 
-    return bestSleeperIds.map(function(sleepNumber) {
+  getWinnerNamesFromList(sortedArray, userRepo) {
```

Comment:
This was a gnarly function. Nice work pulling it out into three separate methods.